src/Native/Runtime/GCHelpers.cpp

```diff
@@ -220,3 +220,63 @@ COOP_PINVOKE_HELPER(void, RhGetMemoryInfo, (UInt32* highMemLoadThreshold, UInt64
                                                        lastRecordedMemLoad,
                                                        lastRecordedHeapSize, lastRecordedFragmentation);
 }
+
+EXTERN_C REDHAWK_API Int64 __cdecl RhGetTotalAllocatedBytes(Boolean precise)
+{
+    if (!precise)
+    {
+#if defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+        uint64_t unused_bytes = Thread::dead_threads_non_alloc_bytes;
+#else
+        // As it could be noticed we read 64bit values that may be concurrently updated.
+        // Such reads are not guaranteed to be atomic on 32bit so extra care should be taken.
+        uint64_t unused_bytes = PalInterlockedCompareExchange64((Int64*)& Thread::dead_threads_non_alloc_bytes, 0, 0);
+#endif
+
+        uint64_t allocated_bytes = GCHeapUtilities::GetGCHeap()->GetTotalAllocatedBytes() - unused_bytes;
```

Comment:
`GetTotalAllocatedBytes` should be called in cooperative mode so that it does not race with the GC.

It may be best to split this in two functions:
- The imprecise one that is `COOP_PINVOKE_HELPER`
- The precise one that is `EXTERN_C`